ast/file.go

```diff
@@ -82,16 +82,10 @@ func NewFileNode(info *FileInfo, syntax *SyntaxNode, decls []FileElement, eof To
 	}
 }
 
+// NewEmptyFileNode returns an empty AST for a file with the given name.
 func NewEmptyFileNode(filename string) *FileNode {
 	fileInfo := NewFileInfo(filename, []byte{})
-	fileInfo.AddToken(0, 0) // EOF
-
-	return &FileNode{
-		compositeNode: compositeNode{
-			children: []Node{NewNoSourceNode(filename)},
```

Comment:
This was the issue: adding a `NoSourceNode` to a `FileNode` was a poor choice. Also, it wasn't even necessary as the replacement code (to just delegate to `NewFileNode`) works and is much more clear.